### PR TITLE
Refine AppWrapper phases and lifecycle

### PR DESCRIPTION
**Pull request by dgrove-oss**
Encode status of child resources via Conditions.
Consolidate and rename Phases.

### CODE REVIEW

File: api/v1beta2/appwrapper_types.go [hunk 1]
```diff
@@ -63,14 +63,21 @@ type AppWrapperStatus struct {
...
+	AppWrapperSucceeded   AppWrapperPhase = "Succeeded"
+	AppWrapperFailed      AppWrapperPhase = "Failed"
+	AppWrapperTerminating AppWrapperPhase = "Terminating"
+)
+
+type AppWrapperCondition string
+
+const (
+	ActiveResources  AppWrapperCondition = "ActiveResources"
+	PassiveResources AppWrapperCondition = "PassiveResources"
```
Comment (tardieu) [inline api/v1beta2/appwrapper_types.go:80]: As named these two conditions sound like opposites, which is not the intent. I suggest we rename the two conditions to make it obvious that:
- one condition is about quota reservation: if false then the appwrapper is not using quota,
- one condition is about deployed resources: if false then there are no wrapped objects deployed on the cluster.

**Review by tardieu — CHANGES_REQUESTED** [review]

File: internal/controller/appwrapper_controller.go [hunk 2]
```diff
@@ -64,6 +64,11 @@ type AppWrapperReconciler struct {
 //+kubebuilder:rbac:groups=cluster.ray.io,resources=rayclusters,verbs=get;list;watch;create;update;patch;delete
 
 // Reconcile reconciles an appwrapper
+// Please see [aw-states] for documentation of this method.
+//
+// [aw-states]: https://github.com/dgrove-oss/appwrapper/blob/crd/docs/state-diagram.md
```
[the remark this answers is not in the document]
Comment (dgrove-oss) [reply internal/controller/appwrapper_controller.go]: changed to project-codeflare.

**Comment by dgrove-oss** [comment]
Ready for another review.

**Review by tardieu — APPROVED** [review]
lgtm

In the "terminating" state, it looks like we could update the two conditions in one api call, but this can be improved in a separate PR.

**Comment by dgrove-oss** [comment]
> lgtm
> 
> In the "terminating" state, it looks like we could update the two conditions in one api call, but this can be improved in a separate PR.

done.

**Pull request merged**